server/src/main/java/org/elasticsearch/index/mapper/NestedObjectMapper.java

```diff
@@ -463,17 +463,14 @@ public boolean hasValue() {
         public void write(XContentBuilder b) throws IOException {
             assert (children != null && children.size() > 0);
             if (children.size() == 1) {
-                b.startObject(leafName());
+                b.field(leafName());
                 leafStoredFieldLoader.advanceTo(children.get(0));
                 leafSourceLoader.write(leafStoredFieldLoader, children.get(0), b);
-                b.endObject();
             } else {
                 b.startArray(leafName());
                 for (int childId : children) {
-                    b.startObject();
                     leafStoredFieldLoader.advanceTo(childId);
                     leafSourceLoader.write(leafStoredFieldLoader, childId, b);
-                    b.endObject();
```

Comment:
These changes here and above are somewhat surprising.. Is the idea that nested objects will be adding their own `start/endObject` tags, so we don't need to do it at the nested object level?

Reply:
Yes, `leafSourceLoader` is responsible for these tags now.